autoload/asyncomplete/sources/tags.vim

```diff
@@ -19,14 +19,18 @@ function! asyncomplete#sources#tags#completor(opt, ctx)
         return
     endif
 
+    let l:tags_tempname = tempname()
+
     let l:matches = {}
 
     for l:tag_file in l:tag_files
-        let l:lines = readfile(l:tag_file)
+        silent exec 'grep! -P "^' . l:kw . '[^\t]*\t" ' . l:tag_file . ' > ' . l:tags_tempname . ' 2>/dev/null'
```

Comment:
Ideally would be good to also support other tools via config like [ripgrep](https://github.com/BurntSushi/ripgrep) but not required for this PR.